plugins/jobs/web_client/views/JobListWidget.js

```diff
@@ -57,27 +79,72 @@ var JobListWidget = View.extend({
 
         eventStream.on('g:event.job_status', this._statusChange, this);
 
-        this.filterTypeMenuWidget = new CheckBoxMenu({
+        this.typeFilterWidget = new CheckBoxMenu({
             title: 'Type',
             values: [],
             parentView: this
         });
 
-        this.filterTypeMenuWidget.on('g:triggerCheckBoxMenuChanged', function (e) {
-            this.typeFilter = _.clone(e);
-            this.render();
+        this.typeFilterWidget.on('g:triggerCheckBoxMenuChanged', function (e) {
+            this.typeFilter = _.keys(e).reduce((arr, key) => {
+                if (e[key]) {
+                    arr.push(key);
+                }
+                return arr;
+            }, []);
+            this._fetchWithFilter();
         }, this);
 
-        this.filterStatusMenuWidget = new CheckBoxMenu({
+        this.statusFilterWidget = new CheckBoxMenu({
             title: 'Status',
             values: [],
             parentView: this
         });
 
-        this.filterStatusMenuWidget.on('g:triggerCheckBoxMenuChanged', function (e) {
-            this.statusFilter = _.clone(e);
-            this.render();
+        let statusTextToStatusCode = {};
+        this.statusFilterWidget.on('g:triggerCheckBoxMenuChanged', function (e) {
+            this.statusFilter = _.keys(e).reduce((arr, key) => {
+                if (e[key]) {
+                    arr.push(parseInt(statusTextToStatusCode[key]));
+                }
+                return arr;
+            }, []);
+            this._fetchWithFilter();
+        }, this);
+
+        restRequest({
+            path: this.showAllJobs ? 'job/meta/all' : 'job/meta',
+            method: 'GET'
+        }).done(result => {
+            var typesFilter = result.types.reduce((obj, type) => {
+                obj[type] = true;
+                return obj;
+            }, {});
+            this.typeFilterWidget.setValues(typesFilter);
+
+            var statusFilter = result.statuses.map(status => {
+                let statusText = JobStatus.text(status);
+                statusTextToStatusCode[statusText] = status;
+                return statusText;
+            }).reduce((obj, statusText) => {
+                obj[statusText] = true;
+                return obj;
+            }, {});
+            this.statusFilterWidget.setValues(statusFilter);
+        });
+
+        this.phaseFilterWidget = new CheckBoxMenu({
```

Comment:
ah, there will have to be a lot of changes throughout the code to update from `phase` to `Timing history`.  Sorry about that!